docs/introduction.rst

```diff
@@ -0,0 +1,67 @@
+==========
+Motivation
+==========
+
+The motivation for Sisyphus was to have framework to easily rerun experiments.
+It should make is simple to come back to an old experiment and see exactly what commands were executed to get the final result.
+Sisyphus makes it easy to have a organized way how to share a workflow, e.g. how to setup a complete translation system from start to end or just use parts of it.
+This is done by creating a graph which connects outputs of jobs (calculations on some input files given some parameter) with other jobs.
+The connections between these jobs are either files or simple python objects.
+
+========
+Concepts
+========
+
+Sisyphus uses a `directed acyclic graph <https://en.wikipedia.org/wiki/Directed_acyclic_graph>`_ to define an experiment workflow.
+The "nodes" of the graph are "Jobs" that are executed on a local machine or in a cluster environments and the edges are physical "files"
+that are passed between jobs.
+This means that all edges leading into the node are the input files to a job, and the outgoing edges are files that the job
+did produce which are passed on to the following jobs.
+In addition to the edges, a job can get "constant" inputs, which are usually fixed parameters that further specify what a certain
+job should do with the input files.
+Each job (instance, not template) in the graph is unique,
+which means that if you try to create a job with the same input files (incoming edges) and the same static parameters as an existing job,
+no new job will be created, but the existing one is re-used.
+Sisyphus uses a hash function over the inputs and static parameters to generate the unique identifier for each job.
+
+It is important to remember that all edges are (physical!) files, without any exception.
+In case that e.g. python variables are passed from one job to another, they are stored as file and then open again in the following job.
+
+A job is divided into tasks, as in order to produce specific output file there can be multiple steps involved.
+Given that all inputs to a job are ready, Sisyphus will try to execute the tasks of a job in the order they are specified,
+by submitting each task to a "job-scheduling-engine", which can be e.g. SGE, slurm or just the local machine.
+Specifying multiple tasks is for example needed if you want to call an external software, and you need to do preparations.
+Then there can be a task ``create_config`` which is executed locally and prepares the needed files,
+and a task ``run`` which calls the software, and is submitted to e.g. a multi-GPU machine when the software needs a lot of resources.
+
+The Sisyphus binary contains three modes: **manager**, **worker** and **console**.
```

Comment:
There is also the shortcut mode, but I think I'll write something about that in a separate commit.